AIFF invalid FORM chunk size fix

Marekkon5 opened the pull request:
Hello, this is a fix to a bug with AIFF I just noticed - I forgot to update the `FORM` chunk header size when the ID3 size changes.
I've also updated the test so it uses temporary file, rather than writing to `testdata` directory.

Sorry for the issues, thanks for the work.

polyfloyd reviewed (changes requested):
Thanks! Could you please add a unit test for this?

Marekkon5 commented:
I could add ffprobe test, since that's how I found this issue, however it is present only in files with actual audio data, the sample AIFF I provided is just minimalistic hand made one without any data other than headers.

polyfloyd commented:
FFprobe is fine, there already is another test that uses it anyway ;) Would it be possible to transcode and commit quiet.mp3 so we can use that?

Marekkon5 commented:
Hello, I've added the `quiet.aiff` file and updated the test to include ffprobe. Also I've changed the API - `read_from_aiff` - wants `io::Read + io::Seek` so it can be used for reading from memory for example. Now `read_from_aiff_file` does the same as the previous API did.

polyfloyd commented on src/tag.rs:
```diff
@@ -1472,8 +1472,15 @@ impl<'a> Tag {
     /// Reads AIFF file and returns ID3 Tag from it
-    pub fn read_from_aiff(path: impl AsRef<Path>) -> crate::Result<Tag> {
-        aiff::load_aiff_id3(path)
+    pub fn read_from_aiff_file(path: impl AsRef<Path>) -> crate::Result<Tag> {
```
Could you please revert the update to the function signature? I have already released it in 0.6.0 and would prefer to not break compatibility. I realize that this is to make it follow the convention of read_from and read_from_path, but I'm afraid it's too late for that.

Marekkon5 replied:
Added in new commit.

polyfloyd commented on src/tag.rs:
```diff
@@ -1472,8 +1472,15 @@ impl<'a> Tag {
+    }
+
+    /// Read ID3 tag from AIFF data in reader
+    pub fn read_from_aiff(reader: &mut (impl io::Read + io::Seek)) -> crate::Result<Tag> {
```
Shall we call this function `read_from_aiff_stream`? Also, the argument type is rather complex. Could we just use `reader: impl io::Read + io::Seek` instead?

Marekkon5 replied:
It throws error that it has to be mutable. `cannot borrow as mutable`

polyfloyd replied:
Sooo, `mut reader`?

Marekkon5 replied:
Sorry, relatively new to Rust, didn't know the syntax was that way too. Added in new commit. Thank you.

polyfloyd replied:
:+1:

polyfloyd reviewed (changes requested):
Excellent! I have left a few more remarks, please have a look :)
